packages/routers/src/http-gateway-routing.ts

```diff
@@ -0,0 +1,92 @@
+import { uriToMultiaddr } from '@multiformats/uri-to-multiaddr'
+import { CID } from 'multiformats/cid'
+import { identity } from 'multiformats/hashes/identity'
+import { fromString as uint8ArrayFromString } from 'uint8arrays/from-string'
+import { toString as uint8ArrayToString } from 'uint8arrays/to-string'
+import type { Provider, Routing, RoutingOptions } from '@helia/interface'
+import { peerIdSymbol, type PeerId, type PeerInfo } from '@libp2p/interface'
+import type { MultihashDigest, Version } from 'multiformats'
+
+export interface HTTPGatwayRouterInit {
+  gateways?: Array<URL | string>
+}
+
+// these values are from https://github.com/multiformats/multicodec/blob/master/table.csv
+const LIBP2P_KEY_CODE = 0x72
```

Comment:
Does it make sense to use the [code for `transport-ipfs-gateway-http` instead](https://github.com/multiformats/multicodec/blob/master/table.csv#L149)?
 
```suggestion
const TRANSPORT_IPFS_GATEWAY_HTTP_CODE = 0x0920
```